flake-module.nix

```diff
@@ -110,6 +110,32 @@ in
                   };
                 };
               };
+              topLevelPackages = mkOption {
```

Comment:
I think the `topLevelPackages` namespace shouldn't be necessary.

Reply:
@roberth Then how do you define `defaultPkg`? Or should you not need to?m

Reply:
We should not need to define a single package.

Currently `haskellProjects.<proj>` defined a flake `packages` package for each `<proj>`. That 1:1 relation is what `package` is for, but now we'll have multiple packages per project.
For each `haskellProjects.<proj>.dirs.*`, we could have a package named `<proj>-<dirname>`.
Ideally we'd use the actual package name instead of the directory name.

A list of directories is actual a bit "high level". I think a good first step is to create an `attrsOf submodule` for the locally defined packages. For example:
`haskellProjects.<proj>.package.<pkgname>.src`
`haskellProjects.<proj>.package.<pkgname>.packageAttribute` (default: `${proj}-${pkgname}`)

The logic for that should be more straightforward to implement than `dirs`, but you can still write `dirs` to make use of those options and get a good separation of concerns. It also allows a bit more control: customizing the package attribute, and perhaps some extra options to pass to cabal2nix.

Reply:
Yes, the structure should be more or less like that. I just put a concrete proposal here (should have done it long ago to assist in this PR): https://github.com/srid/haskell-flake/issues/7#issuecomment-1282932587

The `flakeAttribute` part (or `packageAttribute` in the above comment) is still not clear to me. I suppose we can finalize that once the general functionality is in place (in the PR).